app/controllers/api_controller/parser/request_adapter.rb

```diff
@@ -39,15 +41,16 @@ def c_suffix
       end
 
       def c_id
-        @params[:c_id] || c_path_parts[1]
+        id = @params[:c_id] || c_path_parts[1]
```

Comment:
No, that's because we have allowed for `c_id` to be not-a-number some times. For example here https://github.com/ManageIQ/manageiq/blob/master/spec/requests/api/container_deployments_spec.rb#L5 `c_id` contains `container_deployment_data`.